src/java/org/apache/cassandra/io/sstable/format/SortedTableWriter.java

```diff
@@ -185,9 +197,51 @@ protected boolean startPartitionMetadata(DecoratedKey key, DeletionTime partitio
         return true;
     }
 
+    private void guardCollectionSize(TableMetadata metadata, DecoratedKey partitionKey, Unfiltered unfiltered)
```

Comment:
You can also remove the `metadata` parameter and get the metadata inside this function, closer to where it is needed.